packages/react-aria-components/src/GridList.tsx

```diff
@@ -89,17 +89,8 @@ function GridListInner<T extends object>({props, collection, gridListRef: ref}:
 
   let {gridProps} = useGridList(props, state, ref);
 
-  let children = useCachedChildren({
-    items: collection,
-    children: (item: Node<T>) => {
-      switch (item.type) {
-        case 'item':
-          return <GridListRow item={item} />;
-        default:
-          throw new Error('Unsupported node type in GridList: ' + item.type);
```

Comment:
right, not sure how we should handle that since the collection renderers are supposed to be generic (not specific to one component). Maybe provide a list of allowed child node types?